src/d3a/models/strategy/settlement_strategies/__init__.py

```diff
@@ -0,0 +1,142 @@
+
+        self._update_interval = SettlementTemplateStrategiesConstants.UPDATE_INTERVAL_MIN
+        self.bid_updater = SettlementTemplateStrategyBidUpdater(
+                initial_rate=initial_buying_rate,
+                final_rate=final_buying_rate,
+                fit_to_limit=True,
+                energy_rate_change_per_update=None,
+                update_interval=duration(minutes=self._update_interval),
+                rate_limit_object=min)
+
+        self.offer_updater = SettlementTemplateStrategyOfferUpdater(
+                initial_rate=initial_selling_rate,
+                final_rate=final_selling_rate,
+                fit_to_limit=True,
+                energy_rate_change_per_update=None,
+                update_interval=duration(minutes=self._update_interval),
+                rate_limit_object=max)
+
+    def event_market_cycle(self, strategy):
+        self.bid_updater.update_and_populate_price_settings(strategy.area)
+        self.offer_updater.update_and_populate_price_settings(strategy.area)
+        for market in strategy.area.settlement_markets.values():
+            energy_deviation_kWh = strategy.state.get_unsettled_deviation_kWh(market.time_slot)
+
+            if strategy.state.should_post_bid(market.time_slot):
+                try:
+                    strategy.post_first_bid(
+                        market, energy_deviation_kWh * 1000.0,
+                        self.bid_updater.initial_rate[market.time_slot]
+                    )
+                except MarketException:
+                    pass
+
+            if strategy.state.should_post_offer(market.time_slot):
+                try:
+                    strategy.post_first_offer(
+                        market, energy_deviation_kWh,
+                        self.offer_updater.initial_rate[market.time_slot]
+                    )
+                except MarketException:
+                    pass
+
+    def event_tick(self, strategy):
+        self.bid_updater.increment_update_counter_all_markets(strategy)
+        self.offer_updater.increment_update_counter_all_markets(strategy)
+        """Update posted settlement bids and offers on market tick"""
+        for market in strategy.area.settlement_markets.values():
+            self.bid_updater.update(market, strategy)
+            self.offer_updater.update(market, strategy)
+
+    def _get_settlement_market_by_id(self, area, market_id):
+        markets = [m for m in area.settlement_markets.values() if m.id == market_id]
+        if not markets:
+            return None
+        return markets[0]
+
+    def event_bid_traded(self, strategy, market_id, bid_trade):
+        """Updates the unsettled deviation with the traded energy from the market"""
+        market = self._get_settlement_market_by_id(strategy.area, market_id)
+        if not market:
+            return
+        if bid_trade.offer_bid.buyer == strategy.owner.name:
+            strategy.state.decrement_unsettled_deviation(
```

Comment:
```suggestion
            strategy.state.increment_unsettled_deviation(
```
Assuming the settlement market's bid is following the same approach as normal unsold BID's update rate.

Reply:
This is actually tricky. Unsettled deviation is an absolute value for both consumption and production (bid and offer). Therefore in both cases the value has to be decremented till it reaches zero. Reason why it was modelled this way, to avoid issues with sign convention like the one you mention